subiquity/client/controllers/snaplist.py

```diff
@@ -63,7 +65,11 @@ def cancel(self, sender=None):
         self.app.prev_screen()
 
     async def get_list_wait(self):
-        return await self.endpoint.GET(wait=True)
+        try:
+            return await self.endpoint.GET(wait=True)
+        except aiohttp.client_exceptions.ServerDisconnectedError as sde:
+            log.debug('failed to get snaplist', sde)
+            return SnapListResponse(status=SnapCheckState.FAILED)
```

Comment:
I don't think this is the correct thing really, we should retry on cancellation in the server.